docket_scrap/spiders/docket_spider.py

```diff
@@ -0,0 +1,213 @@
+                    docket["date"] = date
+
+                else:
+                    # if id is not present, then it is merged with previous id
+                    filer = col[0]
+                    filer = self.extract_filer(filer)
+                    docket["filer"] = filer
+
+                    # extracting description from second column
+                    description = col[1]
+                    description = self.extract_description(description)
+                    docket["description"] = description
+
+                    # extracting date from description
+                    date = self.extract_filing_date(description)
+                    docket["date"] = date
+
+                    # copying file_url and id from temporary storage
+                    docket["file_url"] = temporary_storage["file_url"]
+                    docket["docket_id"] = temporary_storage["docket_id"]
+
+            # Since the row length is 1, id and description are merged with previous
+            else:
+
+                # extracting filer from the column
+                filer = col[0]
+                filer = self.extract_filer(filer)
+                docket["filer"] = filer
+
+                # copying previous docket data to current
+                docket["file_url"] = temporary_storage["file_url"]
+                docket["docket_id"] = temporary_storage["docket_id"]
+                docket["description"] = temporary_storage["description"]
+                docket["date"] = temporary_storage["date"]
+
+            # writing docket into csv
+            self.write_in_file(docket)
+
+            yield docket
+
+    def write_in_file(self, docket):
+        # opening file for csv
+        file = open('docket.csv', 'a', encoding='UTF-8', newline='')
+
+        csv_columns = ['docket_id', 'description', 'date', 'filer', 'file_url']
+
+        # creating a csv writer for the file
+        writer = csv.DictWriter(file, csv_columns)
+
+        # generating a row
+        row = dict(docket)
+
+        # writing on csv file
+        writer.writerow(row)
+
+    def extract_anchor(self, docket_id):
+        link = docket_id.css('a::attr("href")').get()
+        if link:
+            if re.match("^.*(.pdf)$", link):
+                link = urljoin(base=self.base_url, url=link)
+            elif re.match('^.*.(.html)$', link):
+                link = urljoin(base=self.base_url, url=link)
+            return link
+        #               Request(url=link, callback=self.fetch_pdf_links)
+        else:
+            return None
+
+    #    def fetch_pdf_links(self, response):
+    # extractor = LinkExtractor(allow='^.*\.(pdf)$')
+    # extractor.extract_links()
+
+    def extract_filing_date(self, description):
+        if description:
+            date = re.search("([0-9]{1,2}/[0-9]{1,2}/[0-9]{2,4})", description)
+            if date:
+                date = date.group()
+                return date
```

Comment:
check id 4892. One of its dates is not extracted. The same has happened with some other entries. You might want to verify what is missing in your regex.